cpp/src/parquet/statistics.cc

```diff
@@ -463,6 +463,8 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
  public:
   using T = typename DType::c_type;
 
+  // Currently, this constructor will tend to be called by ColumnWriter to create the
```

Comment:
I still don't think comment with `currently` and `tend` is a good idea as it may easily be out of sync and confuses users. It would be good enough to simply describe its supposed use case.